spec/controllers/review_controller_spec.rb

```diff
@@ -71,6 +71,72 @@
       end
     end
 
+    describe "POST approve 2i review" do
+      let(:user) { create(:user) }
+      let(:reviewer_user) { create(:user) }
+
+      let(:step_by_step_page) do
+        create(
+          :draft_step_by_step_page,
+          review_requester_id: user.uid,
+          status: "submitted_for_2i"
+        )
+      end
+
+      before do
+        step_by_step_page.update_attributes(:status => 'in_review', :reviewer_id => reviewer_user.uid)
+      end
+
+      it "can be accessed by users with GDS editor, Unreleased feature and 2i reviewer permissions" do
+        stub_user.permissions = ["signin", "GDS Editor", "Unreleased feature", "2i reviewer"]
+        post :approve_2i_review, params: { step_by_step_page_id: step_by_step_page.id }
+
+        expect(response).to redirect_to step_by_step_page_path(step_by_step_page)
+      end
+
+      it "cannot be accessed by users with only GDS editor permissions" do
+        stub_user.permissions << "GDS Editor"
+        post :approve_2i_review, params: { step_by_step_page_id: step_by_step_page.id }
+
+        expect(response.status).to eq(403)
+      end
+
+      it "cannot be accessed by users with only GDS editor and Unreleased feature permissions" do
+        stub_user.permissions = ["signin", "GDS Editor", "Unreleased feature"]
+        post :approve_2i_review, params: { step_by_step_page_id: step_by_step_page.id }
+
+        expect(response.status).to eq(403)
+      end
+
+      it "cannot be accessed by users with only signin permissions" do
+        stub_user.permissions = %w(signin)
+        post :approve_2i_review, params: { step_by_step_page_id: step_by_step_page.id }
+
+        expect(response.status).to eq(403)
+      end
+
+      it "sets status to approved" do
+        stub_user.permissions = ["signin", "GDS Editor", "Unreleased feature", "2i reviewer"]
```

Comment:
Could you extract this out to a `let` variable at the top of this `describe` block?

```ruby
let(:required_permissions) { ["signin", "GDS Editor", "Unreleased feature", "2i reviewer"] }
```

It's used in a few places and if/when it changes in future we should only have to update it once.

Reply:
The combination of users if slightly different in various tests, so I'd rather leave the full list present each time, so we're not switching between different ways of setting the permissions